coalib/parsing/TomlConfParser.py

```diff
@@ -0,0 +1,313 @@
+import os
+import sys
+
+import tomlkit.container
+import tomlkit.items
+from coalib.misc import Constants
+from tomlkit.exceptions import TOMLKitError
+from tomlkit.items import Table, Item
+
+from coalib.results.SourcePosition import SourcePosition
+from coalib.settings.Section import Section
+from coalib.settings.Setting import Setting
+from collections import Iterable, OrderedDict
+import logging
+
+
+class TomlSetting(Setting):
+    """
+    A subclass of a setting but has an additional parameter
+    called the original value. This class is required to
+    infer comments and original data-type of a settings value.
+    """
+
+    def __init__(self, key,
+                 value,
+                 original_value,
+                 origin: (str, SourcePosition) = '',
+                 strip_whitespaces: bool = True,
+                 list_delimiters: Iterable = (',', ';'),
+                 from_cli: bool = False,
+                 remove_empty_iter_elements: bool = True,
+                 to_append: bool = False,
+                 ):
+        self.original_value = original_value
+        super(TomlSetting, self).__init__(
+            key,
+            value,
+            origin,
+            strip_whitespaces,
+            list_delimiters,
+            from_cli,
+            remove_empty_iter_elements,
+            to_append)
+
+
+class TomlConfParser:
+    """
+
+    The parser that generates sections from data in
+    TOML config files. Each TOML table is converted
+    into a section and the all the sections are returned
+    as a ordered dictionary of sections.
+
+    """
+
+    def __init__(self, remove_empty_iter_elements=True):
+
+        self.sections = None
+        self.data = None
+        self.__rand_helper = None
+        self.__init_sections()
+        self.__remove_empty_iter_elements = remove_empty_iter_elements
+        self.logger = logging.getLogger()
+
+    def parse(self, input_data, overwrite=False):
+        """
+        Parses the input and adds the new data to the existing.
+
+        :param input_data: The filename to parse from.
+        :param overwrite:  If True, wipes all existing Settings inside this
```

Comment:
when would you want to overwrite it?

Reply:
I kept the signature of `ConfParser` implementation of parse so that they would . be consistent. But the feature seems to be not used anywhere in coala.
https://github.com/coala/coala/blob/23b9f6204783bc580b9be1060e306af2a0ff6706/coalib/parsing/ConfParser.py#L38